backend/Dockerfile

```diff
@@ -1,8 +1,23 @@
-FROM python:3.12-slim-bullseye
+FROM python:3.12-slim-bullseye AS poetry
 
-RUN pip install "fastapi[standard]" requests
+RUN pip install poetry
 
 RUN mkdir /app
+WORKDIR /app
+
+COPY pyproject.toml /app
+
+FROM poetry as server
+RUN poetry install --no-root
+
 COPY . /app
 
-WORKDIR /app
+RUN poetry install
+
+CMD ["poetry", "run", "fastapi", "dev", "--host", "0.0.0.0", "main.py"]
```

Comment:
We may use prod in Dockerfile and dev in docker-compose.override.yaml

Reply:
poetry run became unnecessary => fixed